### PR TITLE
fix: grants on external volumes

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
Fixes: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2533
Terraform couldn't read privileges for External volumes, because Snowflake returns a shorter `VOLUME` name where we expect `EXTERNAL VOLUME` to be returned. The proposed fix relies on replacing `EXTERNAL VOLUME` with `VOLUME` in the prepare read request function so that the Read operation will be untouched and will work for external volumes.

## Test Plan

* [x] acceptance tests to show the fix works (didn't pass without the fix)

## Other
Wrote to the docs team to add this case to the SHOW GRANTS page.

**Update**: added missing privileges (CREATE MODEL needed for https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2563)

### CODE REVIEW

File: pkg/resources/grant_privileges_to_account_role.go [hunk 1]
```diff
@@ -815,6 +815,11 @@ func prepareShowGrantsRequestForAccountRole(id GrantPrivilegesToAccountRoleId) (
 	case OnAccountObjectAccountRoleGrantKind:
 		data := id.Data.(*OnAccountObjectGrantData)
 		grantedOn = data.ObjectType
+
+		// For EXTERNAL VOLUME, we have to change it to VOLUME because that's what Snowflake returns
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_privileges_to_account_role.go:819]: We have recently fixed similar issue on the SDK level: #2522. Shouldn't we also fix it here on the SDK level (by modifying `func (row grantRow) convert() *Grant {` function)?

File: pkg/resources/grant_privileges_to_account_role_acceptance_test.go [hunk 2]
```diff
@@ -1065,3 +1110,28 @@ func queriedAccountRolePrivilegesContainAtLeast(roleName sdk.AccountObjectIdenti
 		})
 	}, roleName, privileges...)
 }
+
+func createExternalVolumeTemporarily(t *testing.T, externalVolumeName string) {
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_privileges_to_account_role_acceptance_test.go:1114]: Why the `temporarily`?

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/resources/grant_privileges_to_account_role.go (hunk 1 above)
Comment (sfc-gh-jcieslak) [reply pkg/resources/grant_privileges_to_account_role.go:819]: hmmm, true. Done

File: pkg/resources/grant_privileges_to_account_role_acceptance_test.go (hunk 2 above)
Comment (sfc-gh-jcieslak) [reply pkg/resources/grant_privileges_to_account_role_acceptance_test.go]: removed `temporarily` and returned cleanup, so it's consistent with `testint` package utils

File: pkg/resources/grant_privileges_to_account_role.go (hunk 1 above)
Comment (sfc-gh-asawicki) [reply pkg/resources/grant_privileges_to_account_role.go:819]: I don't understand why :(

Comment (sfc-gh-jcieslak) [reply pkg/resources/grant_privileges_to_account_role.go]: I think I was thinking about mapping in a wrong way that's why I thought it couldn't be done, but with mapping VOLUME to EXTERNAL VOLUME can be done in SDK. Moved to the SDK.

**Review by sfc-gh-asawicki — DISMISSED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**